src/BuildScriptGenerator/DefaultDockerfileGenerator.cs

```diff
@@ -0,0 +1,111 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Oryx.BuildScriptGenerator.Exceptions;
+using Microsoft.Oryx.BuildScriptGenerator.Resources;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    internal class DefaultDockerfileGenerator : IDockerfileGenerator
+    {
+        private readonly ICompatiblePlatformDetector _platformDetector;
+        private readonly ILogger<DefaultDockerfileGenerator> _logger;
+
+        private readonly IDictionary<string, IList<string>> _slimPlatformVersions =
+            new Dictionary<string, IList<string>>()
+            {
+                { "dotnetcore", new List<string>() { "2.1" } },
+                { "node",   new List<string>() { "8.16", "10.16" } },
+                { "python", new List<string>() { "3.7" } },
+            };
+
+        private readonly string _templateDockerfile =
+            @"ARG RUNTIME=RUN_IMAGE:RUN_TAG
```

Comment:
move this to the constant file

Reply:
Added a `Dockerfile.oryx.tpl` file that leverages the `TemplateHelper` class and removed this hard-coded string.